### PR TITLE
#10555 added MassConcatenateOutboundPdfs process

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/10555

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java [hunk 1]
```diff
@@ -141,4 +154,28 @@ public I_AD_Archive retrieveArchive(@NonNull final ArchiveId archiveId)
...
+				.map(Optional::get)
+				.map(arch -> AdArchive.builder().id(ArchiveId.ofRepoId(arch.getAD_Archive_ID())).archiveData(archiveBL.getBinaryData(arch)).build());
+	}
+
+	@Override
+	public void updatePrintedRecords(final ImmutableSet<ArchiveId> ids, final UserId userId)
+	{
+		queryBL.createQueryBuilder(I_AD_Archive.class, Env.getCtx(), ITrx.TRXNAME_None)
+				.addOnlyActiveRecordsFilter()
+				.addInArrayFilter(I_AD_Archive.COLUMN_AD_Archive_ID, ids).create().stream().filter(Objects::nonNull).forEach(archive -> archiveEventManager.firePdfUpdate(archive, userId, null));
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java:179]: I guess you don't have access to `X_C_Doc_Outbound_Log_Line.ACTION_PdfExport` in here, right?
Still assuming `null => PdfExport` somewhere in the code is too dirty
So, suggestion: add a method `IArchiveEventManager.firePdfPdfExportUpdate(I_AD_Archive, UserId)`.
Then you can make the action `@NonNull` again

File: backend/de.metas.util/src/main/java/de/metas/util/FileUtil.java [hunk 2]
```diff
@@ -42,14 +41,18 @@ private FileUtil()
 
 	public static final void copy(File from, OutputStream out) throws IOException
 	{
-		InputStream in = new FileInputStream(from);
-		copy(in, out);
+		try (final InputStream in = new FileInputStream(from))
```
Comment (metas-ts) [inline backend/de.metas.util/src/main/java/de/metas/util/FileUtil.java:44]: whoah! thx for finding&fixing

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java:179]: In master `X_C_Doc_Outbound_Log_Line.ACTION_PdfExport` is presumed to be the default action. I guess this is legacy code, where the action is still passed in.
Will add an overload for firePdfUpdate.

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java [hunk 3]
```diff
@@ -141,4 +154,28 @@ public I_AD_Archive retrieveArchive(@NonNull final ArchiveId archiveId)
 	{
 		return InterfaceWrapperHelper.load(archiveId, I_AD_Archive.class);
 	}
+
+	@Override
+	public <T> Stream<AdArchive> streamArchivesForFilter(final IQueryFilter<T> outboundLogFilter, final Class<T> objectClass, final Function<T, TableRecordReference> tableRecordReferenceFunction)
+	{
+		Check.assumeNotNull(outboundLogFilter, "filter cannot not null");
```
[the remark this answers is not in the document]
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java]: Changed to `@NonNull`

File: backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/process/MassConcatenateOutboundPdfs.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		}
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final IQueryFilter<I_C_Doc_Outbound_Log> queryFilter = getProcessInfo()
+				.getQueryFilterOrElse(ConstantQueryFilter.of(false));
+		final Function<I_C_Doc_Outbound_Log, TableRecordReference> logToTableRecordReferenceFunction = log -> TableRecordReference.of(log.getAD_Table_ID(), log.getRecord_ID());
```
[the remark this answers is not in the document]
Comment (adi-stefan) [reply backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/process/MassConcatenateOutboundPdfs.java]: Makes sense, fixed.

File: backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java [hunk 5]
```diff
@@ -141,4 +154,28 @@ public I_AD_Archive retrieveArchive(@NonNull final ArchiveId archiveId)
...
+
+		final IQueryBuilder<T> queryBuilder = queryBL.createQueryBuilder(objectClass)
+				.addOnlyActiveRecordsFilter().filter(outboundLogFilter);
+
+		return queryBuilder.create()
+				.stream()
```
[the remark this answers is not in the document]
Comment (adi-stefan) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/archive/api/impl/ArchiveDAO.java]: did not look at the `stream()` implementation. Fixed.

**Review by metas-ts — APPROVED** [review]

**Pull request merged**